pylint/testutils/functional/find_functional_tests.py

```diff
@@ -38,9 +38,11 @@ def get_functional_test_files_from_directory(
 
     _check_functional_tests_structure(Path(input_dir))
 
-    for dirpath, _, filenames in os.walk(input_dir):
+    for dirpath, dirnames, filenames in os.walk(input_dir):
         if dirpath.endswith("__pycache__"):
             continue
+        dirnames.sort()
+        filenames.sort()
         for filename in filenames:
```

Comment:
```suggestion
        for filename in sorted(filenames):
```

Reply:
Given that we have to use in place sort for `dirnames`, I would prefer to do the same for filenames so that the two `sort()` calls involved in traversal order determinism are grouped.